alembic/env.py

```
@@ -0,0 +1,101 @@
+import asyncio
+from logging.config import fileConfig
+
+from sqlalchemy import pool
+from sqlalchemy.engine import Connection
+from sqlalchemy.ext.asyncio import async_engine_from_config
+
+from alembic import context
+
+from settings import settings
```

Consider performing an early validation right after importing `settings` (or inside `settings`) to fail fast with a clear message rather than letting Alembic later raise a cryptic error. This keeps migrations and app runtime behavior predictable (see checklist item about `settings.DATABASE_URL` usage).